---
title: 无重复RNA-seq差异基因分析
description: 无重复数据差异分析GFOLD, CORNAS, NAISeq等工具
categories:
 - Bioinformation(生物信息)
 - Usage of software(软件工具的使用)
 - Analysis(分析方法)
tags:
 - DEG(差异基因分析)
---

#### 无重复 RNA-seq

无重复RNA-Seq数据差异基因分析


#### NOISeq 2016 (也可用于有重复RNA-seq)
* 原理
    > 1.
    > 
* 材料：
    >+ reads count matrix
    >+ length, GC content,biological classfication...
    ```
    mycount <- reads.tables()
    myfactors <- data.frame(Tissue=c(),TissueRun=c())
    mylength <-
    mygc <-
    mybiotypes <-
    mychrom <- (chr,start,end)

    mydata <- readData(data = mycounts, length = mylength, gc = mygc, biotype = mybiotypes, + chromosome = mychroms, factors = myfactors)
    ```
##### 流程
* 质控
    * Biotype detection
        ```
        mybiodetection <- dat(mydata, k = 0, type = "biodetection", factor = NULL)
        par(mfrow = c(1, 2))
        explo.plot(mybiodetection, samples = c(1, 2), plottype = "persample")

        ```
    * Sequencing depth & Expression Quantiﬁcation
        ```



        ```
    * Sequencing bias detection
        * Length bias
            ```

            ```
        * GC content

        * RNA composition
    
    * PCA
    


    2.标准化，低表达reads过滤
    
    * Normalization (RPKM)
        ```
        myRPKM = rpkm(assayData(mydata)$exprs, long = mylength, k = 0, lc = 1)
        myUQUA = uqua(assayData(mydata)$exprs, long = mylength, lc = 0.5, k = 0)
        myTMM = tmm(assayData(mydata)$exprs, long = 1000, lc = 0)
        ```
    * Low-count ﬁltering (3 methods)
        1.CPM
        2 Wilcoxon test (repeats of each condition ≥5)
        3 Proportion test
        ```
        myfilt = filtered.data(mycounts, factor = myfactors$Tissue, norm = FALSE,  depth = NULL, method = 1/2/3, cv.cutoff = 100, cpm = 1, p.adj = "fdr")
        ```
    * Batch effect
        > ARSyNseq


    3.差异表达分析
    * NOISeq-sim: no replicates available
    ```
     myresults <- noiseq(mydata, factor = "Tissue", k = NULL, norm = "n", pnr = 0.2, + nss = 5, v = 0.02, lc = 1, replicates = "no")
    ```
    
#### GFOLD 2012
* 原理
    > 
>https://zhanglab.tongji.edu.cn/softwares/GFOLD/index.html
* 材料
    >gtf/bed file
    sam file


* 流程
    1. count reads
    >gfold count -ann hg19Ref.gtf -tag sample1.sam -o sample1.read_cnt

    2. Identify differentially expressed genes without replicates
    >gfold diff -s1 sample1 -s2 sample2 -suf .read_cnt -o sample1VSsample2.diff

    2. Identify differentially expressed genes with replicates only in one condition
    >gfold diff -s1 sample1,sample2 -s2 sample3 -suf .read_cnt -o 12VS3.diff

#### CORNAS 2017
* 原理
> Fast Bayesian method   
Generalized Poisson distribution

* 材料
1. 3 tab-separated columns:1. Gene/Transcript/Fragment ID 2. two sample counts 3. columns without headers

2. Configuration file
```
# Parameters to run a Sample A vs Sample B comparison:

# Compulsory parameters:
Gene Name :1
Sample A column: 3
Sample B column: 4

# If coverage is known:
Sample A Coverage: 0.0631
Sample B Coverage: 0.0681 

# With the total read counts for each sample

Alpha: Reducing this percentage will increase sensitivity of making DEG calls. Defualt:99%
Fold thresgold:Reducing this percentage will increase sensitivity of making DEG calls. Default: 1.5

```

###### Run CORNAS
```
Rscript CORNAS.R <config> <datatable>
```
or
```
source("/path/to/CORNAS.R")
cornas("/path/to/config" , "/path/to/datatable")
```

##### Outcome

1. Gene_Name = The identifier of the gene as provided in the input file.
2. DEG_call = Whether the gene is differentially expressed according to the thresholds set: Yes (Y) or No (N).
3. Express_higher = If DEG_call == Y, then which Sample has the higher expression (A or B), otherwise "-".
4. Fold_difference = The fold difference/change of the gene expression between the Sample with higher expression over the other Sample.
5. A_O-count = The observed count in Sample A (as provided in the input file).
6. B_O-count = The observed count in Sample B (as provided in the input file).
7. A_T-lower = The lower bound of the posterior distribution of true counts in Sample A.
8. A_T-upper = The upper bound of the posterior distribution of true counts in Sample A.
9. B_T-lower = The lower bound of the posterior distribution of true counts in Sample B.
10. B_T-upper = The upper bound of the posterior distribution of true counts in Sample B.
